Tests/IntegrationTests/Roslyn/RoslynGenTests.cs

```diff
@@ -92,6 +92,92 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
+            IPackage child1 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child1", "1.1.0", analyzer.GetType().Assembly.Location);
+            IPackage child2 = AddPackage(fakeRemotePkgMgr, "Analyzer.Child2", "1.2.0", analyzer.GetType().Assembly.Location);
+            IPackage targetPkg = AddPackage(fakeRemotePkgMgr, "Empty.Parent", "1.0.0", dummyContentFile, child1, child2);
+
+            string localPackageDestination = TestUtils.CreateTestDirectory(this.TestContext, ".localpackages");
+
+            // Act
+            NuGetPackageHandler nuGetHandler = new NuGetPackageHandler(fakeRemotePkgMgr.LocalRepository, localPackageDestination, logger);
+            AnalyzerPluginGenerator apg = new AnalyzerPluginGenerator(nuGetHandler, logger);
+            ProcessedArgs args = new ProcessedArgs(targetPkg.Id, targetPkg.Version, "cs", null, false, 
+                true /* generate plugins for dependencies with analyzers*/, outputDir);
+            bool result = apg.Generate(args);
+
+            // Assert
+            Assert.IsTrue(result);
+            string[] jarFilePaths = AssertPluginJarsExist(outputDir, 2); // Expecting one plugin per dependency with analyzers
+
+            foreach (string jarFilePath in jarFilePaths)
+            {
+                // Check the content of the files embedded in the jar
+                ZipFileChecker jarChecker = new ZipFileChecker(this.TestContext, jarFilePath);
+
+                // Check the contents of the embedded config file
+                string embeddedConfigFile = jarChecker.AssertFileExists("org\\sonar\\plugins\\roslynsdk\\configuration.xml");
+                RoslynSdkConfiguration config = RoslynSdkConfiguration.Load(embeddedConfigFile);
+
+                // Check the config settings, find which of the dependencies the jar is for (so that we can check the correct strings)
+                IPackage originalPkg = null;
+                if (config.PluginKeyDifferentiator.Equals("analyzerchild1"))
+                {
+                    originalPkg = child1;
+                }
+                if (config.PluginKeyDifferentiator.Equals("analyzerchild2"))
+                {
+                    originalPkg = child2;
+                }
+                Assert.IsNotNull(originalPkg, "Unexpected repository differentiator");
+
+                string pluginId = originalPkg.Id.ToLower();
+                Assert.AreEqual("roslyn." + pluginId + ".cs", config.RepositoryKey, "Unexpected repository key");
+                Assert.AreEqual("cs", config.RepositoryLanguage, "Unexpected language");
+                Assert.AreEqual("dummy title", config.RepositoryName, "Unexpected repository name");
+
+                // Check for the expected property values required by the C# plugin
+                // Property name prefixes should be lower case; the case of the value should be the same as the package id
+                AssertExpectedPropertyDefinitionValue(pluginId + ".cs.analyzerId", originalPkg.Id, config);
```

Comment:
Having to dynamically generate the expected results detracts considerably from the readiability of this test. However, it still offers value by checking that the JARs generated have correctly-associated properties, so I've kept the checks.